src/meshlink.c

```diff
@@ -556,25 +556,25 @@ static bool finalize_join(meshlink_handle_t *mesh) {
     return true;
 }
 
-static bool invitation_send(void *handle, uint8_t type, const void *data, size_t len) {
+static int invitation_send(void *handle, uint8_t type, const void *data, size_t len) {
     meshlink_handle_t* mesh = handle;
     while(len) {
         int result = send(mesh->sock, data, len, 0);
         if(result == -1 && errno == EINTR)
             continue;
         else if(result <= 0)
-            return false;
+            return -1;
         data += result;
         len -= result;
     }
-    return true;
+    return 0;
 }
 
 static bool invitation_receive(void *handle, uint8_t type, const void *msg, uint16_t len) {
     meshlink_handle_t* mesh = handle;
     switch(type) {
         case SPTPS_HANDSHAKE:
-            return sptps_send_record(&(mesh->sptps), 0, mesh->cookie, sizeof mesh->cookie);
+            return !sptps_send_record(&(mesh->sptps), 0, mesh->cookie, sizeof mesh->cookie);
```

Comment:
Yes and I agree there's lots to be documented. It all just returns the send errno if set or -1 when some other error occured, 0 on success.